### ISSUE TITLE
Many 'No diagnostics available' notifcations since 849f544d6 

### ISSUE THREAD

**Issue by n3wborn** [neovim/neovim#21599]
### Neovim version (nvim -v)

NVIM v0.9.0-dev-585+g 849f544d6

### Vim (not Nvim) behaves the same?

no (VIM - Vi IMproved 9.0)

### Operating system/version

Fedora 37

### Terminal name/version

xfce-term

### $TERM environment variable

tmux-256color

### Installation

built from sources

### How to reproduce the issue

```
git checkout 849f544d6~1
make distclean &&
make -j "${ncpu}" CMAKE_BUILD_TYPE="Release"  &&
make install 
nvim
```

nvim opens with an empty buffer, no notification about diagnostic are shown, as usual

```
git checkout 849f544d6
make distclean &&
make -j "${ncpu}" CMAKE_BUILD_TYPE="Release"  &&
make install 
nvim
```

nvim opens with an empty buffer and shows 'No diagnostics available' notification. Many notifications can be shown when opening a non-empty file.

### Expected behavior

nvim should notify  us about diagnostic only when we ask him to do so

### Actual behavior

Nvim spams  'No diagnostics available' notification even on an empty buffer at startup

**Comment by justinmk** [comment]
@will

**Comment by justinmk** [comment]
@leaxoy

**Comment by n3wborn** [comment]
> does reverting [4ace9e7](https://github.com/neovim/neovim/commit/4ace9e7e417fe26c8b73ff1d6042e6e4f3df9ebf) fix the issue?

It does 👍🏽

**Comment by gpanders** [comment]
I think we just need to revert this change:

```lua
  if next(items) == nil then
     vim.notify('No diagnostics available')
     return
   end
```

Using `lwindow` and `cwindow` should be enough to prevent the location/quickfix list from opening when there are no diagnostics.

**Comment by leaxoy** [comment]
I didn't encounter this problem. Have you used the automation function to make multiple calls? I think it only show notification when invoke this function.

**Comment by n3wborn** [comment]
> I didn't encounter this problem. Have you used the automation function to make multiple calls? I think it only show notification when invoke this function.

I don't think so. Not sure to know what this function is about. I just built nvim as I ~daily do

**Comment by kristijanhusak** [comment]
@n3wborn did you manage to solve this somehow?

**Comment by n3wborn** [comment]
> @n3wborn did you manage to solve this somehow?

Nope. for the moment it works because I commented as gpanders says a few messges before

**Comment by ranjithshegde** [comment]
Is there a functional reason as to why this cant exit silently?
I mean, being notified of existing diagnostics itself can be noisy at times, but being notified that there are no diagnostics?

I got used to the older vim workflow of having all the error-related stuff in `qflists`. So I have something like this in my config, so that diagnostics quietly go into loclist _if_ the diagnostics have changed, so that i can inspect them when I need to.
```lua
vim.api.nvim_create_autocmd("DiagnosticChanged", {
  callback = function()
    vim.diagnostic.setloclist({open=false})
  end
})
```
But `DiagnosticChanged` fires even when attaching a client to a buffer, which makes this PR's notify function popup.

For now I have commented out the `vim.notify` call in my runtime file.
